Day-6-7-8-9/src/App.js

```diff
@@ -1,88 +1,89 @@
-import React, { Suspense, useEffect, useState } from "react";
+import React, { Suspense, useContext, useEffect, useState } from "react";
 import ReactDOM from "react-dom/client";
-import { createBrowserRouter, RouterProvider } from "react-router-dom";
-const UserCard = React.lazy(() => import("./components/UserCard"));
+import { createBrowserRouter, Outlet, RouterProvider } from "react-router-dom";
 import Header from "./components/Header";
-import data from "./data/userGithubDetails.json";
 import UserCardInfo from "./components/UserCardInfo";
 import { ErrorComponent } from "./components/ErrorComponent";
 import UserProfile from "./components/UserProfile";
 import UserRepoList from "./components/UserRepoList";
 const SearchComponent = React.lazy(() =>
   import("./components/SearchComponent")
 );
+const UserCard = React.lazy(() => import("./components/UserCard"));
 import AboutUsComponent from "./components/AboutUsComponent";
-import Loader from "./components/Loader";
-import SearchBar from "./components/SearchBar";
-import { fetchUserData } from "./utils/fetchData";
 import ThemeContext from "./components/ThemeContext";
+import Loader from "./components/Loader";
 
-const App = () => {
-  const [memberList, setMemberList] = useState([]);
-
-  useEffect(() => {
-    displayUserData();
-  }, []);
-
-  const displayUserData = () => {
-    const output = [];
-    data &&
-      data.map(async (user) => {
-        const url = `https://api.github.com/users/${user.username}`;
-        const userData = await fetchUserData(url);
-        output.push(userData);
-        setMemberList([...memberList, ...output]);
-      });
-  };
+const Home = () => {
+  const { theme } = useContext(ThemeContext);
 
   return (
-    <div className="home_page">
-      <ThemeContext.Provider value={the}>
-        <Header />
-        <SearchBar
-          users={memberList}
-          searchMembers={setMemberList}
-          fetchUserData={displayUserData}
-        />
-        <Suspense fallback={<Loader />}>
-          <UserCard data={memberList} />
-        </Suspense>
-      </ThemeContext.Provider>
+    <div
+      className={`home_page  ${
+        theme === "light" ? "light_mode" : "dark_mode "
+      }`}
+    >
+      <Header />
+      <div className="body">
+        <Outlet />
+      </div>
     </div>
   );
 };
 
-const router = createBrowserRouter([
+const routeConfig = [
```

Comment:
u can move routes to a separate file, app.js will remain clean.

Reply:
Sure.